src/safeposix/syscalls/sys_calls.rs

```diff
@@ -217,10 +218,40 @@ impl Cage {
         0
     }
 
-    /*
-    *   exec() will only return if error happens 
-    */
+    /// Cleans up all memory mappings from a process's virtual memory space.
+    /// Used by both exec() and exit() to ensure proper memory cleanup,
+    /// but with different purposes:
+    /// - For exec(): Prepares for new process image by clearing old mappings
+    /// - For exit(): Final cleanup before process termination
+    fn clear_memory_mappings(&self) -> Result<(), Errno> {
```

Comment:
I'm confused as to why we need to do this. Shouldn't dropping the VMMap effectively remove this?

Reply:
Yes I think the ownership system might/should drop the cage and and clear its `vmmap`.
ref: https://doc.rust-lang.org/book/ch04-01-what-is-ownership.html#memory-and-allocation

Reply:
IMO it makes sense to just let the vmmap drop on exit() with the rest of the cage struct. For exec we could look into just dropping the vmmap and replacing it with a new one? Seems like it would be more efficient than this.

Reply:
Okay, I have removed the fn and left the code: `vmmap: interface::RustLock::new(Vmmap::new())` in the exec_syscall which should take care for the new cage created